hazelcast/src/test/java/com/hazelcast/config/XmlConfigImportVariableReplacementTest.java

```diff
@@ -411,7 +411,7 @@ public void testImportOverlappingMapConfigFromFile() throws Exception {
         // YAML recursively merges the two files
         assertEquals(1, myMapConfig.getBackupCount());
         MapStoreConfig myMapStoreConfig = myMapConfig.getMapStoreConfig();
-        assertEquals(0, myMapStoreConfig.getWriteDelaySeconds());
+        assertEquals(10, myMapStoreConfig.getWriteDelaySeconds());
```

Comment:
@blazember 

Looks like we might have a conflict of interest here (or at least a grey area).

It turns out that replacing the existing config is actually desired according to https://github.com/hazelcast/hazelcast/pull/14451. If I understand correctly, XML replaces the whole config sections when importing them, and YAML merges them instead

Reply:
@pivovarit You're right, XML replaces the import node with the content of the imported file right in the DOM, while in the case of YAML, the two config files (actually the DOMs built from the two files) need to be merged. The reason for this need is that YAML defines mappings, therefore, it does not allow duplicates, while XML does allow it. The consequence is that the resulting `Config` may be different for the XML and the YAML case if the main and the imported documents have the "same" config (map with the same name for example). There is an ambiguity check though, so actual (scalar) config entries with different values result in failing fast with an `InvalidConfigurationException`. It can be argued though if spreading the "same" configuration across files is a good practice. I think it is not. 

I don't see though how this behavior change impacts config override. We first build a single DOM, then we process it where the overrides can be applied. Is there anything I miss here?

Reply:
Simply put, the test started failing after I introduced the required changes in the parser, and I don't possess enough context to figure out if I'm doing something wrong, or if the test is wrong. Although, I'm leaning towards the latter.

Reply:
I think I see the situation now. So since you changed from `new MapConfig()` to `config.getMapConfig()` in the processor, you get a `MapConfig` with the previously set attributes. So technically, instead of replacing the `MapConfig`, with your change, we start merging in XML too, just like we do with YAML. If I get this right, I agree, the test needs to be modified since we introduce a behavioral change here, which is also a behavioral unification. I wouldn't worry about this since I hardly believe anybody defined the same configuration in two XML files.

Reply:
In such case, we should be ready to go here.